src/parse/pdxfile/lexer.rs

```diff
@@ -442,14 +442,28 @@ impl Iterator for Lexer<'_> {
                     let start_i = i;
                     let start_loc = self.loc;
                     self.consume();
-                    let mut warn_linebreaks = true;
+                    let mut expect_open = false;
+                    let mut expect_immediate_close = false;
+                    let mut immediate_close_warn = false;
                     let mut escaped = false;
                     let mut id = self.start_cob();
                     while let Some((i, c)) = self.peek() {
-                        if c == '[' && i == start_i + 1 {
-                            // A string that starts with `"[` is a datatype expression which might
-                            // be broken into multiple lines for readability.
-                            warn_linebreaks = false;
+                        if c.is_comparator_char() {
+                            // This is a sign that we could be on an intended RHS
+                            expect_open = true;
+                        } else if !c.is_ascii_whitespace() && c != '"' {
+                            expect_open = false;
+                        }
+
+                        if c == ')' {
+                            // Whitespace is not allowed after the closing broce of a quoted trigger
+                            // We should expect to close the quote on the next character or warn
+                            expect_immediate_close = true;
+                        } else if expect_immediate_close && c.is_ascii_whitespace() {
```

Comment:
This might inappropriately trigger in quoted gui datatype expressions, which can have sequences like `)) )` when closing lots of nested parentheses.

Reply:
Localization has its own parser and the format is fully line based, so all line breaks in loca strings have to be `\n`. No need to worry about it here.